packages/airnode-deployer/src/utils/infrastructure.ts

```diff
@@ -34,3 +35,101 @@ export class MultiMessageError extends Error {
     super(messages.join('\n'));
   }
 }
+
+export const BUCKET_NAME_REGEX = /^airnode-[a-f0-9]{12}$/;
+export const generateBucketName = () => `airnode-${randomBytes(6).toString('hex')}`;
+
+export enum FileSystemType {
+  Directory = 'Directory',
+  File = 'File',
+}
+
+export type FileSystemItem = Directory | File;
+
+export type Directory = {
+  type: FileSystemType.Directory;
+  bucketKey: string;
+  children: DirectoryStructure;
+};
+
+export type File = {
+  type: FileSystemType.File;
+  bucketKey: string;
+};
+
+export type DirectoryStructure = Record<string, FileSystemItem>;
+
+export const translatePathsToDirectoryStructure = (paths: String[]) => {
```

Comment:
I'd implement this recursively, but it would be a bit complex either way. 

I think documentation + unit tests are more then enough for this.